task/cargo/0.1/cargo.yaml

```diff
@@ -0,0 +1,40 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: cargo
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "build-tool, rust"
+    tekton.dev/displayName: "Cargo"
+spec:
+  description: >-
+    This task can be used to execute cargo commands for rust projects.
+  workspaces:
+    - name: source
+      description: The workspace that contains your cargo.toml and rust code.
+        This will be the working directory.
+  params:
+    - name: COMMAND
+      type: string
+      description: The cargo command that gets executed.
+        Passed before compile flags. Results in 'cargo $COMMAND' being executed.
+      default: build
+    - name: COMPILE_FLAGS
+      type: array
+      description: Compilation flags passed to cargo.
+        Results in 'cargo $COMMAND $COMPILE_FLAG1 $COMPILE_FLAG2...'.
+      default: []
+    - name: CARGO_IMAGE
+      type: string
+      description: The image that contains cargo.
+        Defaults to the official rust image on the docker hub.
+        When using a custom image, make sure cargo is on the path.
+      default: docker.io/rust@sha256:ea81624bfff9a2d46fc181532acdeafda3cf8c31d25536d443ef902d4894f72a
+  steps:
+    - name: cargo-command
+      image: $(params.CARGO_IMAGE)
+      workingDir: $(workspaces.source.path)
+      command: [cargo]
+      args: ["$(params.COMMAND)", "$(params.COMPILE_FLAGS[*])"]
```

Comment:
a note that providing args directly as args like this makes it so that it might be possible to inject commands you don't want  - might make sense to pass these as env vars to make it harder to do this like in https://github.com/tektoncd/catalog/pull/711 - but that might be tough since COMPILE_FLAGS is an array so mayeb you should just ignore me for now 🤔 

(+ @jerop who has been looking into this)

Reply:
Yes, good catch. I am however unsure how to handle the array aswell. Command could be switched to env but doing just the command and not the compile flags would make the task definition a bit confusing so i'd rather do both at the same time. 


If someone from the team here has a good idea on how to do this I'd happily switch it around to env, if not, maybe this could be added in a later stage?